.github/workflows/ci.yml

```diff
@@ -123,3 +123,52 @@ jobs:
           VALE_VERSION: ${{ env.VALE_VERSION }}
       - name: "Validate documentation style"
         run: ./vale --glob='!{docs/docs-schema-library/reference/*}' $(find ./docs -type f \( -name "*.mdx" -o -name "*.md" \) )
+
+  check-links:
+    if: |
+      always() && !cancelled() &&
+      !contains(needs.*.result, 'failure') &&
+      !contains(needs.*.result, 'cancelled') &&
+      needs.files-changed.outputs.documentation == 'true'
+    needs: ["files-changed", "documentation"]
+    runs-on: "ubuntu-22.04"
+    timeout-minutes: 10
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v4"
+        with:
+          submodules: true
+      - name: "Submodule remote update"
+        run: git submodule update --recursive --remote
+      - name: Install NodeJS
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+          cache-dependency-path: docs/package-lock.json
+      - name: "Install dependencies"
+        working-directory: ./docs
+        run: npm install
+      - name: "Setup Python environment"
+        run: "pip install invoke toml"
+      - name: "Build docs website"
+        working-directory: ./docs
+        run: "invoke docs"
+      - name: "Start Docusaurus server"
+        working-directory: ./docs
+        run: |
+          npm run serve &
+          sleep 10  # Wait for server to start
+      - name: "Check all links"
+        uses: lycheeverse/lychee-action@v2
+        with:
+          args: |
+            --no-progress
+            --config './lychee.toml'
+            --exclude-file './.lycheeignore'
+            --max-cache-age 1h
```

Comment:
I'd put as many of the config options in the config file so its easier to run locally.

When using cache lychee will create a local file to store that cache which would need to be added to .gitignore. If that cache is one hour I don't think we'd ever want to add the file to git.